File: frontend/dental_clinic_app/lib/models/appointment.dart
class Appointment {
  final int? id;
  final int customerId;
  final int employeeId;
  final String? customerName;
  final String? employeeName;
  final DateTime appointmentDateTime;
  final int durationMinutes;
  final AppointmentStatus status;
  final String serviceType;
  final String? description;
  final String? notes;
  final double? cost;

  Appointment({
    this.id,
    required this.customerId,
    required this.employeeId,
    this.customerName,
    this.employeeName,
    required this.appointmentDateTime,
    this.durationMinutes = 60,
    this.status = AppointmentStatus.scheduled,
    required this.serviceType,
    this.description,
    this.notes,
    this.cost,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id'],
      customerId: json['customerId'],
      employeeId: json['employeeId'],
      customerName: json['customerName'],
      employeeName: json['employeeName'],
      appointmentDateTime: DateTime.parse(json['appointmentDateTime']),
      durationMinutes: json['durationMinutes'] ?? 60,
      status: AppointmentStatus.values.firstWhere(
        (e) => e.toString().split('.').last.toUpperCase() == json['status'],
      ),
      serviceType: json['serviceType'],
      description: json['description'],
      notes: json['notes'],
      cost: json['cost']?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'customerId': customerId,
      'employeeId': employeeId,
      'customerName': customerName,
      'employeeName': employeeName,
      'appointmentDateTime': appointmentDateTime.toIso8601String(),
      'durationMinutes': durationMinutes,
      'status': status.toString().split('.').last.toUpperCase(),
      'serviceType': serviceType,
      'description': description,
      'notes': notes,
      'cost': cost,
    };
  }

  DateTime get endTime => appointmentDateTime.add(Duration(minutes: durationMinutes));
}

enum AppointmentStatus { 
  scheduled, 
  confirmed, 
  inProgress, 
  completed, 
  cancelled, 
  noShow 
}

File: backend/src/main/java/com/dentalclinic/entity/Appointment.java
package com.dentalclinic.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "appointments")
public class Appointment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    @NotNull(message = "Customer is required")
    private Customer customer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id", nullable = false)
    @NotNull(message = "Employee is required")
    private Employee employee;

    @NotNull(message = "Appointment date and time is required")
    @Column(name = "appointment_datetime", nullable = false)
    private LocalDateTime appointmentDateTime;

    @Column(name = "duration_minutes")
    private Integer durationMinutes = 60;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AppointmentStatus status = AppointmentStatus.SCHEDULED;

    @NotBlank(message = "Service type is required")
    @Column(name = "service_type")
    private String serviceType;

    @Column(name = "description", length = 500)
    private String description;

    @Column(name = "notes", length = 1000)
    private String notes;

    @Positive(message = "Cost must be positive")
    @Column(name = "cost", precision = 10, scale = 2)
    private BigDecimal cost;

    @OneToMany(mappedBy = "appointment", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Treatment> treatments;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    // Constructors
    public Appointment() {}

    public Appointment(Customer customer, Employee employee, LocalDateTime appointmentDateTime, 
                      String serviceType, String description, BigDecimal cost) {
        this.customer = customer;
        this.employee = employee;
        this.appointmentDateTime = appointmentDateTime;
        this.serviceType = serviceType;
        this.description = description;
        this.cost = cost;
    }

    // Getters and Setters
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public Customer getCustomer() { return customer; }
    public void setCustomer(Customer customer) { this.customer = customer; }

    public Employee getEmployee() { return employee; }
    public void setEmployee(Employee employee) { this.employee = employee; }

    public LocalDateTime getAppointmentDateTime() { return appointmentDateTime; }
    public void setAppointmentDateTime(LocalDateTime appointmentDateTime) { this.appointmentDateTime = appointmentDateTime; }

    public Integer getDurationMinutes() { return durationMinutes; }
    public void setDurationMinutes(Integer durationMinutes) { this.durationMinutes = durationMinutes; }

    public AppointmentStatus getStatus() { return status; }
    public void setStatus(AppointmentStatus status) { this.status = status; }

    public String getServiceType() { return serviceType; }
    public void setServiceType(String serviceType) { this.serviceType = serviceType; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getNotes() { return notes; }
    public void setNotes(String notes) { this.notes = notes; }

    public BigDecimal getCost() { return cost; }
    public void setCost(BigDecimal cost) { this.cost = cost; }

    public List<Treatment> getTreatments() { return treatments; }
    public void setTreatments(List<Treatment> treatments) { this.treatments = treatments; }

    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }

    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }

    public enum AppointmentStatus {
        SCHEDULED, CONFIRMED, IN_PROGRESS, COMPLETED, CANCELLED, NO_SHOW
    }
}